src/tests.rs

```diff
@@ -0,0 +1,203 @@
+        assert_eq!(<EndowedAccount<Test>>::get(), ENDOWED_ID);
+        assert_eq!(Balances::free_balance(&ENDOWED_ID), ENDOWED_BALANCE);
+        // Transfer and check result
+        assert_ok!(Bridge::transfer(Origin::ROOT, 2, 10));
+        assert_eq!(Balances::free_balance(&ENDOWED_ID), ENDOWED_BALANCE - 10);
+        assert_eq!(Balances::free_balance(2), 10);
+    })
+}
+
+#[test]
+fn add_remove_validator() {
+    new_test_ext(1).execute_with(|| {
+        // Already exists
+        assert_noop!(
+            Bridge::add_validator(Origin::signed(VALIDATOR_A), VALIDATOR_A),
+            Error::<Test>::ValidatorAlreadyExists
+        );
+
+        // Errors if added twice
+        assert_ok!(Bridge::add_validator(Origin::signed(VALIDATOR_A), 99));
+        assert_noop!(
+            Bridge::add_validator(Origin::signed(VALIDATOR_A), 99),
+            Error::<Test>::ValidatorAlreadyExists
+        );
+
+        // Confirm removal
+        assert_ok!(Bridge::remove_validator(Origin::signed(VALIDATOR_A), 99));
+        assert_noop!(
+            Bridge::remove_validator(Origin::signed(VALIDATOR_A), 99),
+            Error::<Test>::ValidatorInvalid
+        );
+    })
+}
+
+fn make_proposal(to: u64, amount: u32) -> mock::Call {
+    mock::Call::Bridge(crate::Call::transfer(to, amount))
+}
+
+#[test]
+fn create_sucessful_transfer_proposal() {
+    new_test_ext(2).execute_with(|| {
+        let prop_id: H256 = blake2_256("proposal".as_ref()).into();
+
+        let call = make_proposal(USER, 10);
+
+        assert_eq!(Bridge::validator_threshold(), 2);
+
+        // Create proposal (& vote)
+        assert_ok!(Bridge::create_proposal(
+            Origin::signed(VALIDATOR_A),
+            prop_id.clone(),
+            Box::new(call.clone())
+        ));
+        let prop = Bridge::votes(prop_id.clone()).unwrap();
+        let expected = ProposalVotes {
+            votes_for: vec![VALIDATOR_A],
+            votes_against: vec![],
+            hash: prop_id,
+        };
+        assert_eq!(prop, expected);
+
+        // Second validator votes against
+        assert_ok!(Bridge::vote(Origin::signed(VALIDATOR_B), prop_id, false));
+        let prop = Bridge::votes(prop_id.clone()).unwrap();
+        let expected = ProposalVotes {
+            votes_for: vec![VALIDATOR_A],
+            votes_against: vec![VALIDATOR_B],
+            hash: prop_id,
+        };
+        assert_eq!(prop, expected);
+
+        // Third validator votes in favour
+        assert_ok!(Bridge::vote(Origin::signed(VALIDATOR_C), prop_id, true));
+        let prop = Bridge::votes(prop_id.clone()).unwrap();
+        let expected = ProposalVotes {
+            votes_for: vec![VALIDATOR_A, VALIDATOR_C],
+            votes_against: vec![VALIDATOR_B],
+            hash: prop_id,
+        };
+        assert_eq!(prop, expected);
+
+        assert_eq!(Balances::free_balance(USER), 10);
+        assert_eq!(Balances::free_balance(ENDOWED_ID), ENDOWED_BALANCE - 10);
+    })
```

Comment:
Great test 💯

Reply:
Could you write the same test for receiving assets?

Reply:
Could you elaborate on what you think should be tested?

Reply:
Whether balances change properly when an outgoing transfer happens. As you mentioned, that is part of another pallet though – so not needed here.